File: java_solutions/811.subdomain-visit-count.java
/*
 * @lc app=leetcode id=811 lang=java
 *
 * [811] Subdomain Visit Count
 */

record Solution() {
    public java.util.List<String> subdomainVisits(String[] cpdomains) {
        return java.util.Arrays.stream(cpdomains).flatMap(s -> java.util.Optional.of(s.split(" ")).map(parts -> java.util.stream.IntStream.range(0, parts[1].split("\\.").length).mapToObj(i -> java.util.Arrays.stream(parts[1].split("\\.")).skip(i).collect(java.util.stream.Collectors.joining("."))).map(sub -> new Object[]{sub, Integer.parseInt(parts[0])})).get()).collect(java.util.stream.Collectors.groupingBy(obj -> (String)obj[0], java.util.stream.Collectors.summingInt(obj -> (int)obj[1]))).entrySet().stream().map(e -> e.getValue() + " " + e.getKey()).toList();
    }
}

File: kotlin_solutions/811.subdomain-visit-count.kt
/*
 * @lc app=leetcode id=811 lang=kotlin
 *
 * [811] Subdomain Visit Count
 */

class Solution { fun subdomainVisits(cpdomains: Array<String>): List<String> = mutableMapOf<String, Int>().apply { cpdomains.forEach { s -> s.split(" ").let { (cnt, dom) -> dom.split(".").indices.forEach { i -> dom.split(".").drop(i).joinToString(".").let { sub -> this[sub] = getOrDefault(sub, 0) + cnt.toInt() } } } } }.map { "${it.value} ${it.key}" } }